pkg/api/disruption_report.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	apitype "github.com/openshift/sippy/pkg/apis/api"
+	bqcachedclient "github.com/openshift/sippy/pkg/bigquery"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+func GetDisruptionVsPrevGAReportFromBigQuery(client *bqcachedclient.Client) (apitype.DisruptionReport, []error) {
+	generator := disruptionReportGenerator{
+		client:   client.BQ,
+		viewName: "BackendDisruptionPercentilesDeltaCurrentVsPrevGA",
+	}
+
+	return getReportFromCacheOrGenerate[apitype.DisruptionReport](client.Cache, generator, generator.GenerateReport, apitype.DisruptionReport{})
```

Comment:
I don't think the caching will work, getReportFromCacheOrGenerate will use the JSON marshalled `generator` as the cache key, but there's no public fields in generator so it'll probably just marshal to `{}` (getReportFromCacheORGenerate should probably check for that).

Reply:
Ok added, and the cache will error if it sees an empty json struct